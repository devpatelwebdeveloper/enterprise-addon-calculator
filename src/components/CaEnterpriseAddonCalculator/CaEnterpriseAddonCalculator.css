:root {
  --breakpointsSmall: @theme breakpointsSmall;
  --breakpointsMedium: @theme breakpointsMedium;
}

.CaEnterpriseAddonCalculator {
  @media screen and (min-width: var(--breakpointsSmall)) {
    padding: 20px;
  }
}

.users {
  display: flex;
  justify-content: center;
}

.radio {
  position: relative;
  display: flex;
  align-items: center;
  line-height: 1;
  margin-right: 8px;
  margin-bottom: 40px;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    margin-right: 40px;
  }
}

.radio-label {
  cursor: pointer;
  padding-left: 8px;
}

.radio-input {
  position: absolute;
  width: 20px;
  height: 20px;
  opacity: 0;
  cursor: pointer;

  &:checked {
    & ~ .radio-circle {
      &:after {
        content: "";
        display: block;
        height: 12px;
        width: 12px;
        background: @theme colorPrimary;
        border-radius: 50%;
      }
    }
  }
}

.radio-circle {
  cursor: pointer;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: solid 1px #00aeef;
  background: #ffffff;
  display: flex;
  justify-content: center;
  align-items: center;
}

.chart {
  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    flex-wrap: nowrap;
  }
}

.chart-products {
  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    background: #ffffff;
    border: solid 1px #e3e5e8;
    padding: 20px;
    border-radius: 4px;
    display: flex;
    max-width: 843px;
    margin-bottom: 20px;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    margin-bottom: 0;
  }
}

/* Individual Price Cards */
.chart-item {
  background: #ffffff;
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: 8px;
  border: solid 1px #f4f5f8;
  border-radius: 4px;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    max-width: 280px;
    padding: 0 20px;
    border: none;
    border-right: solid 1px #f4f5f8;
    border-radius: none;
    margin-bottom: 20px;

    &:last-child {
      padding: 0 0 0 20px;
      border-right: none;
    }

    &:first-child {
      padding: 0 20px 0 0;
    }
  }
}

.chart-item-top {
  margin-bottom: 20px;
  padding-bottom: 20px;
  display: flex;
  position: relative;
  width: 100%;

  &:after {
    content: "";
    width: 40px;
    height: 2px;
    border-radius: 2px;
    background: @theme colorPrimary;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

.chart-item-title {
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 600;
  letter-spacing: normal;
  width: 100%;
  flex: 1 1 156px;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 1.5rem;
    line-height: 1.75rem;
    font-weight: 600;
    letter-spacing: normal;
    height: 56px;
  }
}

.chart-toggle {
  position: relative;
}

.chart-toggle-label {
  width: 52px;
  height: 30px;
  background: #6b6c72;
  border-radius: 30px;
  position: relative;
  border: none;
  cursor: pointer;
  display: block;

  &:after {
    position: absolute;
    content: "";
    display: block;
    height: 24px;
    width: 24px;
    border-radius: 50%;
    background: #ffffff;
    left: 3px;
    top: 3px;
    transition: left 0.2s cubic-bezier(0.35, 0, 0.25, 1);
  }
}

.chart-toggle-input {
  position: absolute;
  width: 52px;
  height: 30px;
  opacity: 0;

  &:checked ~ .chart-toggle-label {
    background: @theme colorPrimary;

    &:after {
      left: calc(100% - 3px - 24px);
    }
  }
}

.chart-price-add {
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  font-weight: 700;
  margin-bottom: 4px;
}

.chart-price {
  display: flex;
  align-items: flex-start;
  line-height: 1;
  margin-bottom: 20px;
  padding-bottom: 20px;
  border-bottom: solid 1px #f4f5f8;
}

.chart-price-dollars {
  font-size: 1.75rem;
  line-height: 2rem;

  font-weight: 600;
  line-height: 1;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 2.25rem;
    line-height: 2.75rem;
  }
}

.chart-price-cents {
  font-size: 1rem;
  margin-top: 3px;
  line-height: 1;
  font-weight: 700;
}

.chart-price-term {
  font-size: 1rem;
  font-weight: 400;
  font-weight: 700;
  line-height: 1;
  align-self: flex-end;
  margin-bottom: 4px;
}

.chart-features {
  list-style-type: none;
  margin: 0 0 20px 0;
  padding: 0;
}

.chart-features-item {
  margin-bottom: 8px;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
  letter-spacing: normal;
  display: flex;

  /* :global .checkmark {
      width: 11px;
      height: 11px;
      flex: 0 0 11px;
      right: 0;
      margin-right: $xs-spacing;
      top: 0;

      &::before {
        height: 100%;
      }

      &::after {
        width: 60%;
      }
    } */
}

.chart-button {
  width: auto;
  margin-top: auto;
}

.total {
  background: #ffffff;
  border: solid 1px #e3e5e8;
  padding: 20px;
  border-radius: 4px;
  width: 100%;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    display: flex;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    display: block;
    max-width: 268px;
    margin-left: 20px;
  }
}

.total-top {
  flex: 0 0 50%;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    border-right: solid 1px #f4f5f8;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    border-right: none;
  }
}

.total-title {
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 600;
  letter-spacing: normal;
  padding-bottom: 8px;
  margin-bottom: 40px;
  position: relative;

  &:after {
    content: "";
    width: 40px;
    height: 2px;
    border-radius: 2px;
    background: @theme colorPrimary;
    position: absolute;
    bottom: 0;
    left: 0;
  }

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

.total-price {
  display: flex;
  align-items: flex-start;
  line-height: 1;
}

.total-price-dollars {
  font-size: 2.25rem;
  line-height: 2.75rem;
  letter-spacing: normal;
  font-weight: 600;
  line-height: 1;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 2.75rem;
  }
}

.total-price-cents {
  font-size: 1.25rem;
  font-weight: 600;
  letter-spacing: normal;
  margin-top: 3px;
  line-height: 1;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 1.5rem;
  }
}

.total-price-term {
  font-size: 1.25rem;
  font-weight: 600;
  line-height: 1;
  align-self: flex-end;
  margin-bottom: 4px;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    font-size: 1.5rem;
  }
}

.total-price-info {
  font-size: 0.875rem;
  line-height: 1.25rem;
  letter-spacing: normal;
  font-weight: 500;
  margin: 20px 0;
  border-bottom: solid 1px #f4f5f8;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    border-bottom: none;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    border-right: none;
    padding-bottom: 20px;
    border-bottom: solid 1px #f4f5f8;
  }
}

.includes {
  flex: 0 0 50%;

  /* TABLET STYLES */
  @media screen and (min-width: var(--breakpointsSmall)) {
    padding-left: 20px;
  }

  /* DESKTOP STYLES */
  @media screen and (min-width: var(--breakpointsMedium)) {
    padding-left: 0px;
  }
}

.includes-title {
  font-size: 0.875rem;
  line-height: 1.25rem;
  margin-bottom: 20px;
  font-weight: 500;
}

.includes-item {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
  margin-bottom: 20px;
  display: flex;
  justify-content: space-between;
}

.includes-cost {
  font-weight: 500;
}

/* MOBILE STYLES */
@media screen and (max-width: var(--breakpointsMedium) - 1) {
}

/* DESKTOP STYLES */
@media screen and (min-width: var(--breakpointsMedium)) {
}

/* TABLET STYLES */
@media screen and (min-width: var(--breakpointsSmall)) {
}
