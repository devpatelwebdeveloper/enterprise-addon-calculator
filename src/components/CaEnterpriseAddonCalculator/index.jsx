import React, { createContext, useReducer, useContext, useState } from "react";
import classNames from "classnames/bind";
import styles from "./CaEnterpriseAddonCalculator.css";

// Constants
const featuresData = [
  {
    name: "inventory",
    title: "Advanced inventory",
    features: [
      "Prioritize sales orders and improved picking process",
      "Save time and effort by making reviewing more automated",
      "Easily track product movement and storage"
    ],
    ctas: [
      {
        text: "Learn more",
        url: "https://quickbooks.intuit.com/ca/enterprise/advanced-inventory/"
      }
    ]
  },
  {
    name: "pricing",
    title: "Advanced pricing",
    features: [
      "Set sophisticated price rules",
      "Create scheduled promotions with future start dates",
      "Automate your Pricing"
    ],
    ctas: [
      {
        text: "Learn more",
        url: "https://quickbooks.intuit.com/ca/enterprise/advanced-pricing/"
      }
    ]
  },
  {
    name: "hosting",
    title: "Hosting",
    features: [
      "Run Enterprise from anytime, anywhere, on any device",
      "Easily collaborate with your teams by accessing the same information at the same time",
      "Data is securely backed up daily"
    ],
    ctas: [
      {
        text: "Learn more",
        url: "https://quickbooks.intuit.com/ca/enterprise/cloud-hosting/"
      }
    ]
  }
];

const costData = [
  {
    users: 5,
    cost: 380,
    features: {
      inventory: {
        state: false,
        cost: 106
      },
      pricing: {
        state: false,
        cost: 43
      },
      hosting: {
        state: false,
        cost: 253
      }
    }
  },
  {
    users: 10,
    cost: 569,
    features: {
      inventory: {
        state: true,
        cost: 0
      },
      pricing: {
        state: true,
        cost: 0
      },
      hosting: {
        state: false,
        cost: 506
      }
    }
  },
  {
    users: 15,
    cost: 759,
    features: {
      inventory: {
        state: true,
        cost: 0
      },
      pricing: {
        state: true,
        cost: 0
      },
      hosting: {
        state: false,
        cost: 759
      }
    }
  },
  {
    users: 20,
    cost: 918,
    features: {
      inventory: {
        state: true,
        cost: 0
      },
      pricing: {
        state: true,
        cost: 0
      },
      hosting: {
        state: false,
        cost: 1012
      }
    }
  },
  {
    users: 25,
    cost: 1107,
    features: {
      inventory: {
        state: true,
        cost: 0
      },
      pricing: {
        state: true,
        cost: 0
      },
      hosting: {
        state: false,
        cost: 1265
      }
    }
  },
  {
    users: 30,
    cost: 1265,
    features: {
      inventory: {
        state: true,
        cost: 0
      },
      pricing: {
        state: true,
        cost: 0
      },
      hosting: {
        state: false,
        cost: 1518
      }
    }
  }
];

const initState = {
  features: {
    pricing: false,
    inventory: false,
    hosting: false
  },
  total: 0,
  users: 5,
  costing: costData[0]
};

const store = createContext();

const { Provider } = store;

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer((state, action) => {
    const { type, payload } = action;

    switch (type) {
      case "setInventory":
        return {
          ...state,
          features: { ...state.features, inventory: payload }
        };
      case "setPricing":
        return { ...state, features: { ...state.features, pricing: payload } };
      case "setHosting":
        return { ...state, features: { ...state.features, hosting: payload } };
      case "setTotal":
        return { ...state, total: payload };
      case "setUsers":
        return { ...state, users: payload };
      case "setCosting":
        return { ...state, costing: payload };
    }
  }, initState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

const EnterprisePackageItem = ({ cost, name, title, features, ctas }) => {
  const cx = classNames.bind(styles);
  const { state, dispatch } = useContext(store);
  const setType = `set${name.charAt(0).toUpperCase() + name.slice(1)}`;

  const handleChange = (event) => {
    dispatch({ type: setType, payload: event.currentTarget.checked });
  };

  return (
    <div className={cx("chart-item")}>
      <div className={cx("chart-item-top")}>
        <div className={cx("chart-item-title")}>{title}</div>

        <div className={cx("chart-toggle")}>
          <input
            data-testid={`${name}-toggle`}
            className={cx("chart-toggle-input")}
            checked={state.features[name]}
            onChange={(event) => handleChange(event)}
            id={`${name}-toggle`}
            name={name}
            type="checkbox"
            value={name}
            disabled={state.costing.features[name].state === true}
          />
          <label
            className={cx("chart-toggle-label")}
            htmlFor={`${name}-toggle`}
          ></label>
        </div>
      </div>

      <div className={cx("chart-price-add")}>Add</div>
      <div className={cx("chart-price")}>
        <div className={cx("chart-price-dollars")}>${cost}</div>
        <div className={cx("chart-price-cents")}>00</div>
        <div className={cx("chart-price-term")}>/mo.</div>
      </div>

      <ul className={cx("chart-features", "checkmark__list")}>
        {features.map((feature, index) => (
          <li key={index} className={cx("chart-features-item")}>
            <span className="checkmark"></span> {feature}
          </li>
        ))}
      </ul>
      {ctas &&
        ctas.map((cta, index) => (
          <button key={index} url={cta.url} className={cx("chart-button")}>
            {cta.text}
          </button>
        ))}
    </div>
  );
};

const EnterprisePackageTotal = ({ base }) => {
  const cx = classNames.bind(styles);
  const globalState = useContext(store);
  const { state, dispatch } = globalState;

  const features = featuresData;

  const featuresState = state.features;
  const featuresCost = state.costing.features;

  const getFeatureTitle = (name) => {
    const filtered = features.filter((feature) => feature.name === name);

    return filtered[0].title;
  };

  const getFeatureList = () => {
    let features = [];
    for (const feature in featuresState) {
      if (featuresState[feature] === true) {
        let featureObj = {};
        featureObj.text = getFeatureTitle(feature);
        featureObj.cost = featuresCost[feature].cost;

        features.push(featureObj);
      }
    }
    return features;
  };

  const featureList = getFeatureList();

  const reducer = (accumulator, currentValue) =>
    accumulator + currentValue.cost;

  const total = featureList.reduce(reducer, state.costing.cost);

  return (
    <div className={cx("total")}>
      <div className={cx("total-top")}>
        <div className={cx("total-title")}>Total</div>
        <div className={cx("total-price")}>
          <div data-testid="total" className={cx("total-price-dollars")}>
            ${total}
          </div>
          <div className={cx("total-price-cents")}>00</div>
          <div className={cx("total-price-term")}>/mo.</div>
        </div>
        <div className={cx("total-price-info")}>
          Annual subscription billed monthly
        </div>
      </div>
      <div className={cx("includes")}>
        <div className={cx("includes-title")}>Your subscription includes:</div>
        <div className={cx("includes-item")}>
          <div className={cx("includes-text")}>
            Desktop Enterprise Subscription
          </div>
          <div className={cx("includes-cost")}>${base}</div>
        </div>
        {(featureList === undefined || featureList.length !== 0) &&
          featureList.map((feature, index) => (
            <div key={index} className={cx("includes-item")}>
              <div className={cx("includes-text")}>{feature.text}</div>
              <div className={cx("includes-cost")}>${feature.cost}</div>
            </div>
          ))}
      </div>
    </div>
  );
};

const EnterprisePackageBuilder = () => {
  const cx = classNames.bind(styles);
  const selectedUsers = (users) => {
    return costData.filter((item) => item.users === users)[0];
  };

  const globalState = useContext(store);
  const { state, dispatch } = globalState;

  const handleChange = (event) => {
    const num = parseInt(event.currentTarget.value);
    const userCost = selectedUsers(num);

    for (const feature in userCost.features) {
      const setType = `set${
        feature.charAt(0).toUpperCase() + feature.slice(1)
      }`;
      const featState = userCost.features[feature].state;

      if (feature === "hosting" && state.features.hosting === true) {
        featState = true;
      }

      dispatch({ type: setType, payload: featState });
    }

    dispatch({ type: "setUsers", payload: num });

    dispatch({ type: "setCosting", payload: userCost });
  };

  const features = featuresData;

  return (
    <div
      data-testid="CaEnterpriseAddonCalculator"
      data-tracking="ca_enterprise_addon_calculator"
      className={cx("CaEnterpriseAddonCalculator")}
    >
      {/* USERS */}
      <div className={cx("users")}>
        {costData.map((item) => (
          <div className={cx("radio")} key={item.users}>
            <input
              data-testid={`users-${item.users}`}
              className={cx("radio-input")}
              id={`users-${item.users}`}
              name="users"
              type="radio"
              value={item.users}
              checked={state.users === item.users}
              onChange={handleChange}
            ></input>
            <div className={cx("radio-circle")}></div>
            <label
              className={cx("radio-label")}
              htmlFor={`users-${item.users}`}
            >
              {item.users}
            </label>
          </div>
        ))}
      </div>

      {/* END USERS */}

      <div className={cx("chart")}>
        <div className={cx("chart-products")}>
          {features.map((feature) => {
            return (
              <EnterprisePackageItem
                key={feature.name}
                name={feature.name}
                title={feature.title}
                cost={state.costing.features[feature.name].cost}
                features={feature.features}
                ctas={feature.ctas}
              />
            );
          })}
        </div>

        <EnterprisePackageTotal base={state.costing.cost} />
      </div>
    </div>
  );
};

const CaEnterpriseAddonCalculator = () => (
  <StateProvider>
    <EnterprisePackageBuilder />
  </StateProvider>
);

CaEnterpriseAddonCalculator.propTypes = {};

export default CaEnterpriseAddonCalculator;
