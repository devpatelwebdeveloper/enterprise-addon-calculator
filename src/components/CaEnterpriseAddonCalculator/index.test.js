import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import CaEnterpriseAddonCalculator from "./index";

afterEach(cleanup);

describe("CaEnterpriseAddonCalculator", () => {
  test("renders correctly", () => {
    const { getByTestId } = render(<CaEnterpriseAddonCalculator />);
    expect(getByTestId("CaEnterpriseAddonCalculator")).toMatchSnapshot();
  });
  test("updates total", () => {
    const { getByTestId } = render(<CaEnterpriseAddonCalculator />);
    expect(getByTestId("total").innerHTML).toBe("$380");
    fireEvent.click(getByTestId("users-10"));
    expect(getByTestId("total").innerHTML).toBe("$569");
  });
  test("updates total with toggle", () => {
    const { getByTestId } = render(<CaEnterpriseAddonCalculator />);
    expect(getByTestId("total").innerHTML).toBe("$380");
    fireEvent.click(getByTestId("inventory-toggle"));
    expect(getByTestId("total").innerHTML).toBe("$486");
  });
});
