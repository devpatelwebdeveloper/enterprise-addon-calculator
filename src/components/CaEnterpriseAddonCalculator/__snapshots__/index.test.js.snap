// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaEnterpriseAddonCalculator renders correctly 1`] = `
<DocumentFragment>
  <div
    class="CaEnterpriseAddonCalculator"
    data-testid="CaEnterpriseAddonCalculator"
    data-tracking="ca_enterprise_addon_calculator"
  >
    <div
      class="users"
    >
      <div
        class="radio"
      >
        <input
          checked=""
          class="radio-input"
          data-testid="users-5"
          id="users-5"
          name="users"
          type="radio"
          value="5"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-5"
        >
          5
        </label>
      </div>
      <div
        class="radio"
      >
        <input
          class="radio-input"
          data-testid="users-10"
          id="users-10"
          name="users"
          type="radio"
          value="10"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-10"
        >
          10
        </label>
      </div>
      <div
        class="radio"
      >
        <input
          class="radio-input"
          data-testid="users-15"
          id="users-15"
          name="users"
          type="radio"
          value="15"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-15"
        >
          15
        </label>
      </div>
      <div
        class="radio"
      >
        <input
          class="radio-input"
          data-testid="users-20"
          id="users-20"
          name="users"
          type="radio"
          value="20"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-20"
        >
          20
        </label>
      </div>
      <div
        class="radio"
      >
        <input
          class="radio-input"
          data-testid="users-25"
          id="users-25"
          name="users"
          type="radio"
          value="25"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-25"
        >
          25
        </label>
      </div>
      <div
        class="radio"
      >
        <input
          class="radio-input"
          data-testid="users-30"
          id="users-30"
          name="users"
          type="radio"
          value="30"
        />
        <div
          class="radio-circle"
        />
        <label
          class="radio-label"
          for="users-30"
        >
          30
        </label>
      </div>
    </div>
    <div
      class="chart"
    >
      <div
        class="chart-products"
      >
        <div
          class="chart-item"
        >
          <div
            class="chart-item-top"
          >
            <div
              class="chart-item-title"
            >
              Advanced inventory
            </div>
            <div
              class="chart-toggle"
            >
              <input
                class="chart-toggle-input"
                data-testid="inventory-toggle"
                id="inventory-toggle"
                name="inventory"
                type="checkbox"
                value="inventory"
              />
              <label
                class="chart-toggle-label"
                for="inventory-toggle"
              />
            </div>
          </div>
          <div
            class="chart-price-add"
          >
            Add
          </div>
          <div
            class="chart-price"
          >
            <div
              class="chart-price-dollars"
            >
              $106
            </div>
            <div
              class="chart-price-cents"
            >
              00
            </div>
            <div
              class="chart-price-term"
            >
              /mo.
            </div>
          </div>
          <ul
            class="chart-features checkmark__list"
          >
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Prioritize sales orders and improved picking process
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Save time and effort by making reviewing more automated
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Easily track product movement and storage
            </li>
          </ul>
          <div
            class=""
          >
            <a
              class="Button_button__328896b9 Button_standard__328896b9 Button_primary__328896b9 chart-button"
              data-action="interacted"
              data-testid="button"
              data-tracking="button"
              data-wa-link="learn-more-cta-enterprise-calculator-inventory"
              href="https://quickbooks.intuit.com/ca/enterprise/advanced-inventory/"
              title=""
            >
              <span
                class="Button_buttonText__328896b9"
              >
                Learn more
              </span>
            </a>
          </div>
        </div>
        <div
          class="chart-item"
        >
          <div
            class="chart-item-top"
          >
            <div
              class="chart-item-title"
            >
              Advanced pricing
            </div>
            <div
              class="chart-toggle"
            >
              <input
                class="chart-toggle-input"
                data-testid="pricing-toggle"
                id="pricing-toggle"
                name="pricing"
                type="checkbox"
                value="pricing"
              />
              <label
                class="chart-toggle-label"
                for="pricing-toggle"
              />
            </div>
          </div>
          <div
            class="chart-price-add"
          >
            Add
          </div>
          <div
            class="chart-price"
          >
            <div
              class="chart-price-dollars"
            >
              $43
            </div>
            <div
              class="chart-price-cents"
            >
              00
            </div>
            <div
              class="chart-price-term"
            >
              /mo.
            </div>
          </div>
          <ul
            class="chart-features checkmark__list"
          >
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Set sophisticated price rules
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Create scheduled promotions with future start dates
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Automate your Pricing
            </li>
          </ul>
          <div
            class=""
          >
            <a
              class="Button_button__328896b9 Button_standard__328896b9 Button_primary__328896b9 chart-button"
              data-action="interacted"
              data-testid="button"
              data-tracking="button"
              data-wa-link="learn-more-cta-enterprise-calculator-pricing"
              href="https://quickbooks.intuit.com/ca/enterprise/advanced-pricing/"
              title=""
            >
              <span
                class="Button_buttonText__328896b9"
              >
                Learn more
              </span>
            </a>
          </div>
        </div>
        <div
          class="chart-item"
        >
          <div
            class="chart-item-top"
          >
            <div
              class="chart-item-title"
            >
              Hosting
            </div>
            <div
              class="chart-toggle"
            >
              <input
                class="chart-toggle-input"
                data-testid="hosting-toggle"
                id="hosting-toggle"
                name="hosting"
                type="checkbox"
                value="hosting"
              />
              <label
                class="chart-toggle-label"
                for="hosting-toggle"
              />
            </div>
          </div>
          <div
            class="chart-price-add"
          >
            Add
          </div>
          <div
            class="chart-price"
          >
            <div
              class="chart-price-dollars"
            >
              $253
            </div>
            <div
              class="chart-price-cents"
            >
              00
            </div>
            <div
              class="chart-price-term"
            >
              /mo.
            </div>
          </div>
          <ul
            class="chart-features checkmark__list"
          >
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Run Enterprise from anytime, anywhere, on any device
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Easily collaborate with your teams by accessing the same information at the same time
            </li>
            <li
              class="chart-features-item"
            >
              <span
                class="checkmark"
              >
                <img
                  alt=""
                  src="https://quickbooks.intuit.com/oidam/intuit/sbseg/en_ca/shared/icons/checkmark-thick.svg"
                />
              </span>
               Data is securely backed up daily
            </li>
          </ul>
          <div
            class=""
          >
            <a
              class="Button_button__328896b9 Button_standard__328896b9 Button_primary__328896b9 chart-button"
              data-action="interacted"
              data-testid="button"
              data-tracking="button"
              data-wa-link="learn-more-cta-enterprise-calculator-hosting"
              href="https://quickbooks.intuit.com/ca/enterprise/cloud-hosting/"
              title=""
            >
              <span
                class="Button_buttonText__328896b9"
              >
                Learn more
              </span>
            </a>
          </div>
        </div>
      </div>
      <div
        class="total"
      >
        <div
          class="total-top"
        >
          <div
            class="total-title"
          >
            Total
          </div>
          <div
            class="total-price"
          >
            <div
              class="total-price-dollars"
              data-testid="total"
            >
              $380
            </div>
            <div
              class="total-price-cents"
            >
              00
            </div>
            <div
              class="total-price-term"
            >
              /mo.
            </div>
          </div>
          <div
            class="total-price-info"
          >
            Annual subscription billed monthly
          </div>
        </div>
        <div
          class="includes"
        >
          <div
            class="includes-title"
          >
            Your subscription includes:
          </div>
          <div
            class="includes-item"
          >
            <div
              class="includes-text"
            >
              Desktop Enterprise Subscription
            </div>
            <div
              class="includes-cost"
            >
              $380
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
